### PR TITLE
Minor connections refactor

### PR DESCRIPTION
**Pull request by code-asher**
More details in the commit messages.

### CODE REVIEW

File: lib/vscode/src/vs/server/node/connection.ts [hunk 1]
```diff
@@ -181,20 +212,20 @@ export class ExtensionHostConnection extends Connection {
...
 					break;
 				case 'VSCODE_EXTHOST_DISCONNECTED':
-					this.logger.trace('Going offline');
+					this.logger.debug('Got disconnected message');
 					this.setOffline();
 					break;
 				case 'VSCODE_EXTHOST_IPC_READY':
-					this.logger.trace('Got ready message');
-					this.sendInitMessage(buffer);
+					this.logger.debug('Handshake completed');
```
Comment (jsjoeio) [inline lib/vscode/src/vs/server/node/connection.ts:219]: Nice addition 🎉

File: lib/vscode/src/vs/server/node/protocol.ts [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 export interface SocketOptions {
+	/** The token is how we identify and connect to existing sessions. */
 	readonly reconnectionToken: string;
+	/** Specifies that the client is trying to reconnect. */
 	readonly reconnection: boolean;
+	/** If true assume this is not a web socket (always false for code-server). */
 	readonly skipWebSocketFrames: boolean;
+	/** Whether to support compression (web socket only). */
 	readonly permessageDeflate?: boolean;
+	/**
```
Comment (jsjoeio) [inline lib/vscode/src/vs/server/node/protocol.ts:17]: awesome docs! thanks for adding 👏

**Review by jsjoeio — APPROVED** [review]
This is awesome! Love the code comments too ♥️

P.S. - was this the thing we were discussing where I said I would learn from the code and write a test?

**Comment by code-asher** [comment]
Fixed up the double/single quote errors.

Nope, this is mostly just to set the stage so when it happens again we'll have some better logs to figure out what's going on.

File: lib/vscode/src/vs/server/node/connection.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ export abstract class Connection {
 	) {
 		this.logger = logger.named(
 			this.name,
-			field("token", this.protocol.options.reconnectionToken),
+			field('token', this.protocol.options.reconnectionToken),
```
Comment (jawnsy) [inline lib/vscode/src/vs/server/node/connection.ts:30]: what's the difference between `"` and `'` in JavaScript/TypeScript?

Comment (code-asher) [reply lib/vscode/src/vs/server/node/connection.ts:30]: None but VS Code's linting rules enforce `'` for "unlocalized" strings so the only time they are allowed is in calls like:

```
localize('projectQuickPick.help', "TypeScript help")
```

Comment (jawnsy) [reply lib/vscode/src/vs/server/node/connection.ts:30]: ah very interesting pattern, so they're using something that extracts all `"`-quoted strings into resource bundles?

Comment (code-asher) [reply lib/vscode/src/vs/server/node/connection.ts:30]: I think they might be doing something like that, although if they can already detect and report to the dev when a string should be using `"` or `'` I'm not sure why they can't just use that knowledge to extract it and don't bother the dev. :stuck_out_tongue:

**Comment by jsjoeio** [comment]
Nice work! Merge whenever 🚢

**Pull request merged**